packages/components/view-switcher/src/view-switcher-button.tsx

```diff
@@ -7,52 +7,66 @@ import {
 import { css } from '@emotion/react';
 import AccessibleButton from '@commercetools-uikit/accessible-button';
 import { designTokens } from '@commercetools-uikit/design-system';
+import { warning } from '@commercetools-uikit/utils';
 import { getButtonStyles } from './view-switcher.styles';
 
 export type TViewSwitcherButtonProps = {
-  isDisabled?: boolean;
+  children?: string;
+  icon?: ReactElement;
   isActive?: boolean;
-  children: string;
+  isDisabled?: boolean;
   isCondensed?: boolean;
   isFirstButton?: boolean;
   isLastButton?: boolean;
-  icon?: ReactElement;
+  label?: string;
   value: string;
   onClick?: (
     event: MouseEvent<HTMLButtonElement> | KeyboardEvent<HTMLButtonElement>
   ) => void;
 };
 
-const ViewSwitcherButton = (props: TViewSwitcherButtonProps) => (
-  <AccessibleButton
-    label={props.children}
-    css={getButtonStyles(
-      props.isDisabled,
-      props.isActive,
-      props.isCondensed,
-      props.isFirstButton,
-      props.isLastButton
-    )}
-    onClick={props.onClick}
-    isDisabled={props.isDisabled}
-  >
-    {props.icon && (
-      <span
-        css={css`
-          margin: ${designTokens.marginForViewSwitcherIcon};
-          display: flex;
-          align-items: center;
-          justify-content: center;
-        `}
-      >
-        {cloneElement(props.icon, {
-          size: props.isCondensed ? 'medium' : 'big',
-        })}
-      </span>
-    )}
-    <span>{props.children}</span>
-  </AccessibleButton>
-);
+const ViewSwitcherButton = (props: TViewSwitcherButtonProps) => {
+  warning(
+    Boolean(props.children || props.icon),
+    'uikit/ViewSwitcherButton: You need to provide at least the `children` to render inside the button or an `icon`.'
+  );
+
+  warning(
+    Boolean(props.label || props.children),
+    'uikit/ViewSwitcherButton: You need to provide a `label` when rendering only an `icon` within the button.'
+  );
+
+  return (
+    <AccessibleButton
+      label={props.label ?? props.children ?? ''}
```

Comment:
Note to self: `children` can only only be a string.

Reply:
What do we need to do about that?